src/anomalib/models/image/vlm/lightning_model.py

```diff
@@ -0,0 +1,82 @@
+"""Visual Anomaly Model for Zero/Few-Shot Anomaly Classification."""
+
+# Copyright (C) 2024 Intel Corporation
+# SPDX-License-Identifier: Apache-2.0
+
+import logging
+from enum import Enum
+
+import torch
+from torch.utils.data import DataLoader
+
+from anomalib import LearningType
+from anomalib.models import AnomalyModule
+
+from .backends import Backend, Ollama
+
+logger = logging.getLogger(__name__)
+
+
+class VLMBackend(Enum):
+    """Supported VLM backends."""
+
+    OLLAMA = "ollama"
+
+
+class Vlm(AnomalyModule):
+    """Visual anomaly model."""
+
+    def __init__(self, backend: VLMBackend = VLMBackend.OLLAMA, api_key: str | None = None, k_shot: int = 3) -> None:
```

Comment:
quite minor, but it would be good if `backend` could be str as well

Reply:
I was thinking of an alternative where instead of the llm provider, the users can select the model, and based on it the model will select the right provider.